frontend/src/views/NewShoot.vue

```diff
@@ -516,28 +521,33 @@ export default {
     }
   },
   async beforeRouteLeave (to, from, next) {
-    if (to.name === 'NewShootEditor') {
-      if (!this.valid) {
-        if (!await this.confirmNavigateToYamlIfInvalid()) {
-          return next(false)
+    if (this.sortedCloudProviderKindList.length) {
+      if (to.name === 'NewShootEditor') {
+        if (!this.valid) {
+          if (!await this.confirmNavigateToYamlIfInvalid()) {
+            return next(false)
+          }
         }
-      }
-      await this.updateShootResourceWithUIComponents()
-      return next()
-    } else {
-      if (!this.isShootCreated && await this.isShootContentDirty()) {
-        if (!await this.confirmNavigation()) {
-          return next(false)
+        await this.updateShootResourceWithUIComponents()
+      } else {
+        if (!this.isShootCreated && await this.isShootContentDirty()) {
+          if (!await this.confirmNavigation()) {
+            return next(false)
+          }
```

Comment:
avoid deep if-nestings
```js
async beforeRouteLeave(to, from, next) {
  if (!this.sortedCloudProviderKindList.length) {
    return next()
  }

  if (to.name === "NewShootEditor") {
    if (!this.valid && !await this.confirmNavigateToYamlIfInvalid()) {
      return next(false)
    }
    
    await this.updateShootResourceWithUIComponents()
    return next()
  }

  if (!this.isShootCreated && await this.isShootContentDirty() && !await this.confirmNavigation()) {
    return next(false)
  }

  return next()
}
```
should behave the same. But please double check :)

Reply:
I adapted it myself and came up with the same code so I guess it will work